src/modules/commander/Commander.cpp

```diff
@@ -835,10 +983,10 @@ Commander::handle_command(const vehicle_command_s &cmd)
 			}
 
 			if (cmd.param1 > 0.5f) {
-				res = main_state_transition(_status, commander_state_s::MAIN_STATE_OFFBOARD, _status_flags, &_internal_state);
+				res = try_mode_change(commander_state_s::MAIN_STATE_OFFBOARD, false, true);
```

Comment:
I think we should probably just leave the NAV_GUIDED_ENABLE case alone as it's already a bit convoluted.

Reply:
I am not sure i fully get your point, i replaced `main_state_transition()` to `try_mode_change() `here is to get rid of the duplicated `print_reject_mode("OFFBOARD");`, there is no functional change here :thinking: